test/metabase/test/data/dataset_definition_test.clj

```diff
@@ -0,0 +1,32 @@
+(ns metabase.test.data.dataset-definition-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.driver :as driver]
+   [metabase.driver.ddl.interface :as ddl.i]
+   [metabase.test :as mt]
+   [toucan2.core :as t2]))
+
+(deftest dataset-with-custom-pk-test
+  (mt/test-drivers (mt/sql-jdbc-drivers)
+    (mt/dataset (mt/dataset-definition "custom-pk"
+                  ["user"
+                   [{:field-name "uuid" :base-type :type/UUID :pk? true}]
+                   [[#uuid "201014e3-81bc-4349-a64f-d5519571297b"]]]
+                  ["group"
+                   [{:field-name "user_uuid" :base-type :type/UUID :fk "user"}]
+                   [[#uuid "201014e3-81bc-4349-a64f-d5519571297b"]]])
+      (let [user-fields  (t2/select [:model/Field :name :semantic_type :fk_target_field_id] :table_id (mt/id :user))
+            group-fields (t2/select [:model/Field :name :semantic_type :fk_target_field_id] :table_id (mt/id :group))]
+        (testing "user.uuid is PK"
+          (is (= [{:name               (ddl.i/format-name driver/*driver* "uuid")
```

Comment:
nit I would break out a `format-name` helper function which closes over the driver in the let block above, as you call this 3 times.

Reply:
did that [98b3216](https://github.com/metabase/metabase/pull/39038/commits/98b3216404e50c36c56ccb53a61f6255b39245fe)